include/openssl/core.h

```diff
@@ -25,6 +25,11 @@ extern "C" {
  * to communicate data between them.
  */
 
+/* Opaque handles to be used with core upcall functions */
+typedef void *OSSL_CORE_HANDLE;
```

Comment:
I think this is a wrong way how to declare an opaque handle as it won't provide type safety. The right way IMO is:

`typedef struct ossl_core_handle OSSL_CORE_HANDLE;`

Reply:
Oh, now that I see this, I would advice against pointer typedefs. They can be very confusing at times. And I do agree with @t8m.

Reply:
Fixed

Reply:
@mattcaswell You've probably forgot to push the change?